src/server/pfs/server/driver.go

```diff
@@ -1729,14 +1755,86 @@ func (d *driver) squashCommitSet(txnCtx *txncontext.TransactionContext, commitse
 			return &pfsserver.ErrSquashWithoutChildren{Commit: ci.Commit}
 		}
 	}
-
 	if err := d.squashCommitSetInternal(txnCtx, commitInfos); err != nil {
 		return err
 	}
+	for _, ci := range commitInfos {
+		txnCtx.StopJob(ci.Commit)
+	}
+	return nil
+}
 
-	// notify PPS that this commitset has been squashed so it can clean up any
-	// jobs associated with it at the end of the transaction
-	txnCtx.StopJobs(commitset)
+// getCommitSubvenance gets the subvenance of the provided commit.
+// The provided commit is included in the returned list.
+// There are no guarantees about the ordering of the returned list.
+func (d *driver) getCommitSubvenance(txnCtx *txncontext.TransactionContext, commit *pfs.Commit) ([]*pfs.CommitInfo, error) {
+	commitInfos, err := d.inspectCommitSetImmediate(txnCtx, &pfs.CommitSet{ID: commit.ID})
+	if err != nil {
+		return nil, err
+	}
+	// Create an index from branch -> commit for direct provenance lookups.
+	branchIndex := make(map[string]*pfs.CommitInfo)
+	for _, ci := range commitInfos {
+		branchIndex[pfsdb.BranchKey(ci.Commit.Branch)] = ci
+	}
+	visited := make(map[string]*pfs.CommitInfo)
+	// Traverse will fill the visited map with commits that are traversed.
+	// Non-nil commits are in the subvenance.
+	var traverse func(*pfs.CommitInfo)
+	traverse = func(commitInfo *pfs.CommitInfo) {
+		if _, ok := visited[pfsdb.CommitKey(commitInfo.Commit)]; ok {
+			return
+		}
+		if proto.Equal(commitInfo.Commit, commit) {
+			visited[pfsdb.CommitKey(commitInfo.Commit)] = commitInfo
+			return
+		}
+		visited[pfsdb.CommitKey(commitInfo.Commit)] = nil
+		for _, provBranch := range commitInfo.DirectProvenance {
+			ci := branchIndex[pfsdb.BranchKey(provBranch)]
+			traverse(ci)
+			ci = visited[pfsdb.CommitKey(ci.Commit)]
+			if ci != nil {
+				visited[pfsdb.CommitKey(commitInfo.Commit)] = commitInfo
+				return
+			}
+		}
+	}
+	for _, ci := range commitInfos {
+		traverse(ci)
+	}
+	var result []*pfs.CommitInfo
+	for _, ci := range visited {
+		if ci != nil {
```

Comment:
when would `ci == nil`?

Reply:
When a commit isn't in the subvenance of the commit being squashed.

Reply:
reading it over I don't see how that would occur

Reply:
That happens when multiple commits are created in the same transaction.